lib/ace/transport_app.rb

```diff
@@ -28,6 +29,61 @@ def initialize(config = nil)
       super(nil)
     end
 
+    def self.init_puppet_target(device_json)
+      begin
+        device = JSON.parse(device_json)
+      rescue StandardError
+        # Resource JSON error
+        # Do not add error to log as potentially includes passwords
+        ACE::Error.new('There was an error parsing the json of the Puppet target',
+                       'ace/puppet-target-error')
+        return nil
+      end
+
+      unless device['target']
+        ACE::Error.new("There was an error parsing the Puppet target. 'target' not found",
+                       'ace/puppet-target-error')
+        return nil
+      end
+      unless device['target']['remote-transport']
+        ACE::Error.new("There was an error parsing the Puppet target. 'remote-transport' not found",
+                       'ace/puppet-target-error')
+        return nil
+      end
+      if !device['target']['uri'] && !device['target']['host']
```

Comment:
This, and some of the parsing code below needs to be replaced by the `body['target']['host'] || body['target']['name']` bit from https://github.com/puppetlabs/ace/blob/5019cabcaf1eb9d0242792df09c4cbfb97a971e1/lib/ace/transport_app.rb#L68-L71 - see the comments there for explanation.

@Thomas-Franklin had exactly the same issue this morning. Please get together and have a `certname_from_target(target)` method to do the lookup, so when bolt inevitably fixes this, we only have a single place to change.

Reply:
We rather use compiler/certname for our name here, as discussed?

Reply:
https://github.com/puppetlabs/ace/pull/26/files#diff-5bb6dbadea974c974791e62f1392e908R118 is what I have done for the trusted facts

Reply:
After the slack convo, I was assuming that I was just now passing through the compiler certname, would that need validation at this point?

Reply:
> We rather use compiler/certname for our name here, as discussed?

yes

> would that need validation at this point?

no